haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -195,13 +196,30 @@ def train(
         :param train_loss: Specify the training loss to use to fit the Sentence-Transformers model. Possible options are
             "mnrl" (Multiple Negatives Ranking Loss) and "margin_mse".
         :param num_workers: The number of subprocesses to use for the Pytorch DataLoader.
-        :param use_amp: Use Automatic Mixed Precision (AMP).
+        :param use_amp: Whether to use Automatic Mixed Precision (AMP).
+        :param gradient_checkpointing: Whether to use gradient checkpointing for the HuggingFace Transformers model
+            that is loaded by Sentence Transformers. Gradient checkpointing saves on GPU memory at the expenses of
+            slower training times. A general rule of thumb is that gradient checkpointing slows down training by about 20%.
         :param kwargs: Additional training keyword arguments to pass to the `SentenceTransformer.fit` function. Please
             reference the Sentence-Transformers [documentation](https://www.sbert.net/docs/training/overview.html#sentence_transformers.SentenceTransformer.fit)
             for a full list of keyword arguments.
         """
         send_event(event_name="Training", event_properties={"class": self.__class__.__name__, "function_name": "train"})
 
+        try:
+            from sentence_transformers.models import Transformer
+        except (ImportError, ModuleNotFoundError) as ie:
+            from haystack.utils.import_utils import _optional_component_not_installed
+
+            _optional_component_not_installed(__name__, "sentence", ie)
```

Comment:
Hey @sjrl, first off,  this trick unfortunately won't work anymore since we introduced generalimport in Haystack. However I'm a bit confused about this block on a more basic level. Why is there an optional import, given that you added no additional dependencies? Also, Haystack has no `sentence` extra as far as I'm aware. Can you elaborate a bit what were you trying to do here?

Reply:
Hey @ZanSara I need access to the `Transformer` class from `sentence_transformers` and I was following the example that was used earlier in the init function in this class, which you can see here https://github.com/deepset-ai/haystack/blob/ba06bc4805630ae8e1c4fd7abde6383afcbde78e/haystack/nodes/retriever/_embedding_encoder.py#L112-L121

I'm happy to remove this optional import and just do a normal one at the beginning of this file, but I was mostly confused by the fact that this optional import already exists in the above link. 

> Also, Haystack has no sentence extra as far as I'm aware. Can you elaborate a bit what were you trying to do here?

I was following the code example I linked above so it looks like the `optional` install should be updated there as well.

Reply:
I removed the code I added for the optional import and I removed the (I guess forgotten) optional import code in the init method and moved it as a normal import to the top of the file. Please let me know if this is alright.

Reply:
Yes that's good, thank you! This looked quite like a very old leftover :+1: